torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -354,6 +354,178 @@ def layer_norm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+
+    assert isinstance(normalized_shape, (int, Sequence, torch.Size))
+    normalized_size = normalize_to_torch_size(normalized_shape)
+    input_ndim = input_strategy.output_ndim
+    axis = input_ndim - len(normalized_size)
+    outer_dims = list(range(axis))
+
+    assert isinstance(output_mask, List) and len(output_mask) == 3
+
+    # output triple: (d_input, d_weight, d_bias)
+    out_tuple_strategy = OpStrategy([])
+    for idx, _ in enumerate(input_strategy.strategies):
+        output_specs_list: List[Optional[DTensorSpec]] = []
+        op_args_target_specs = []
+        redistribute_costs = []
+        input_src_spec = input_strategy.strategies[idx].output_spec
+        assert isinstance(input_src_spec, DTensorSpec)
+
+        # arg: grad_out
+        # TODO: change the strategy to the following rule.
+        # d_input is basically a product of element-wise mul of
+        # grad_out, rstd, and normalized input, among which rstd
+        # and normalized input (x_hat) should have the same sharding
+        # placements, and grad_out's sharding is determined by the
+        # pointwise result of x_hat and weight/bias.
+        if output_mask[0]:
+            # TODO: now grad_out spec follows input spec. we may need
+            # to change it to apply a pointwise rule over grad_out,
+            # input, and weight.
+            grad_out_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(input_src_spec.placements, axis),
+                tensor_meta=input_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(grad_out_target_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(grad_out_strategy, grad_out_target_spec)
+            )
+            output_specs_list.append(grad_out_target_spec)
+            out_tuple_strategy.strategies.append(
+                # these 3 lists will be filled in later
+                PlacementStrategy(
+                    output_spec=output_specs_list,
+                    input_specs=op_args_target_specs,
+                    redistribute_cost=redistribute_costs,
+                )
+            )
+        else:
+            output_specs_list.append(None)
+
+        # arg: input
+        input_target_spec = DTensorSpec(
+            mesh=mesh,
+            placements=_replicate_dims_start_at(input_src_spec.placements, axis),
+            tensor_meta=input_src_spec.tensor_meta,
+        )
+        op_args_target_specs.append(input_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(input_strategy, input_target_spec)
+        )
+
+        # arg: mean, rstd
+        mean_target_spec = mean_strategy.strategies[idx].output_spec
+        assert isinstance(mean_target_spec, DTensorSpec)
+        op_args_target_specs.append(mean_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(mean_strategy, mean_target_spec)
```

Comment:
given that we don't change the mean_target_spec at all, the redistribute_cost should simply be 
[0.0] * len(mean_strategy.strategies)?

Reply:
I see. This means we do not pass down the `redistribute_cost` along the computation chain? For example, ((a + b) * c), the redistribute cost from `+` won't be passed to `*`.

Reply:
yes we don't need to pass down redistribute cost along the chain, even with the current change I believe the redistribute_cost generated by this function call are all 0s